src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -100,19 +93,81 @@ public void ResolveFileProviders(IConfiguration configuration)
                 WebRootFileProvider = new PhysicalFileProvider(WebRootPath);
             }
 
-            if (this.IsDevelopment())
-            {
-                StaticWebAssetsLoader.UseStaticWebAssets(this, configuration);
-            }
         }
 
         public string ApplicationName { get; set; }
         public string EnvironmentName { get; set; }
 
-        public IFileProvider ContentRootFileProvider { get; set; }
-        public string ContentRootPath { get; set; }
+        private IFileProvider _contentRootFileProvider = NullFileProvider;
+        public IFileProvider ContentRootFileProvider
+        {
+            get => _contentRootFileProvider;
+            set
+            {
+                if (_contentRootPath == default(string))
+                {
+                    throw new InvalidOperationException("Can only set ContenRootFileProvider if ContentRootPath is set.");
+                }
+                _contentRootFileProvider = value;
+            }
+        }
+        private IFileProvider _webRootFileProvider = NullFileProvider;
+        public IFileProvider WebRootFileProvider
+        {
+            get => _webRootFileProvider;
+            set
+            {
+                if (_contentRootPath == default(string))
+                {
+                    throw new InvalidOperationException("Can only set ContenRootFileProvider if ContentRootPath is set.");
+                }
+                _webRootFileProvider = value;
+            }
+        }
+
+        // ContentRootPath and WebRootPath are set to default! on
+        // initialization to match the behavior in HostingEnvironment.
+        private string _contentRootPath = default!;
+        public string ContentRootPath
+        {
+            get => _contentRootPath;
+            set
+            {
+                _contentRootPath = value;
+                if (Directory.Exists(_contentRootPath))
+                {
+                    _contentRootFileProvider = new PhysicalFileProvider(_contentRootPath);
+                } 
+            }
+        }
+
+        private string _webRootPath = default!;
+        public string WebRootPath
+        {
+            get => _webRootPath;
+            set
+            {
+                _webRootPath = ResolveWebRootPath(value, ContentRootPath ?? Directory.GetCurrentDirectory());
+                if (Directory.Exists(_webRootPath))
+                {
+                    _webRootFileProvider = new PhysicalFileProvider(_webRootPath);
+                }
+            }
+        }
+
+        private string ResolveWebRootPath(string webRootPath, string contentRootPath)
+        {
+            if (string.IsNullOrEmpty(webRootPath))
+            {
+                return Path.GetFullPath(contentRootPath);
```

Comment:
If you call `IWebHostBuilder.UseSetting(WebHostDefaults.WebRootKey, null)` with the generic host does the web root become the content root? This seems very dangerous because we don't know what sensitive files might be in the content root that might now be served by static files or something. We should definitely have test cases for these kind of scenarios.

Reply:
Updated this to no-op if the user attempts to set the `WebRootPath` to null. This matches the behavior in the current `GenericHostBuilder` as well.